Docs 2146 business events v4.2 (#893)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/common-dev-strategies.adoc

> == Implement Enterprise Integration Patterns

Reviewer: Lines 7
To help you develop your Mule runtime engine applications, consider common development strategies and practices such as creating reproducible builds; modularizing your configuration files; implementing reliability patterns, transaction management, continuous integration, orchestration in Mule, and enterprise integration patterns; and configuring business events tracking.

Reviewer: Lines 8-15
== Create Reproducible Builds
Create reproducible builds to facilitate application maintenance by running a particular version of your Mule projects at a time. You can achieve this goal by:
* Using source control for your Mule projects
* Managing dependencies by using Apache Maven and an in-house manager +

For details, see xref:reproducible-builds.adoc[Reproducible Builds] documentation.

Reviewer: == Implement Reliability Patterns
Implement a reliability pattern in your application design so your application is capable of reliable messaging, even if receives messages from a nontransactional connector. A reliability pattern design couples a reliable acquisition flow with an application logic flow. +
For details, see xref:reliability-patterns.adoc[Reliability Patterns] documentation.

Reviewer: == Implement Transaction Management
Implement transaction operations in a Mule app for which the result cannot remain indeterminate. When a series of steps in a flow must succeed or fail as one unit, Mule uses a transaction to demarcate that unit. +
For details on different transaction types, transaction actions operations, and how you can configure a transaction in a message source or Try Scope, see the xref:transaction-management.adoc[Transaction Management] documentation. +
Additionally, you can implement xref:single-resource-transaction.adoc[single-source transactions], xref:xa-transactions.adoc[XA transactions], and xref:using-bitronix-to-manage-transactions.adoc[Britonix to manage transactions].

Reviewer: == Modularize Configurations Files
Modularize configuration files by splitting monolithic configurations into several files, and leverage the Mule capacity to load multiple configuration files at application startup. 
Although it might seem convenient to keep your Mule configurations in one file, a very large XML file quickly becomes unmanageable. Modularizing Mule configurations enables you to split work across teams, whether these teams work on the same project or on different projects that share the same work. 
You can load multiple configuration files in either of two ways:

* Side-by-side
Provide a list of independent configuration files to load.
* Imported
Have one configuration file import several others that, in turn, can import other files. +
For details see the xref:modularizing-your-configuration-files-for-team-development.adoc[Modularize Configurations Files] documentation.

Reviewer: == Implement Continuous Integration
Implement continuous integration (CI) for your projects by using Apache Maven as your build tool. Create a build that gets triggered on every project change, and run all of its unit and functional tests automatically so that you can:
* Get early notification of issues in the software development lifecycle
* Ensure that code is fully tested before release
* Confirm that successfully tested branches ensure better success when merging to the master branch. +
For details, see the xref:continuous-integration.adoc[Continuous Integration] documentation.

Reviewer: == Implement Orchestration in Mule
Implement orchestration in Mule to facilitate moving data through a Mule implementation based on flows. Orchestration implements business-level processes combining business-specific services across applications and information systems.
Flows use cases that orchestrate processes include:
* Simple integration
* Scheduling data processing
* Connecting cloud and on-premise applications
* Event processing in which multiple services must be composed and must operate simultaneously.+
For details, see the xref:understanding-orchestration-using-mule.adoc[Orchestration Using Mule] documentation.

> Implement enterprise integration patterns as solutions for designing, building messaging and integration systems, as well as to provide a common language for teams to use when architecting solutions. +

Reviewer: ```suggestion
Implement enterprise integration patterns as solutions for designing, building, messaging, and integration systems, as well as to provide a common language for teams to use when architecting solutions. +
```
BUT...does this text belong here or in the previous topic?

> For details, see xref:understanding-enterprise-integration-patterns-using-mule.adoc[Enterprise Integration Patterns Using Mule] documentation.

Reviewer: ```suggestion
For details, see the xref:understanding-enterprise-integration-patterns-using-mule.adoc[Enterprise Integration Patterns Using Mule] documentation.
```

== Configure Business Events Tracking

> Mule applications can collect business event information about flows and message processors that handle your business transactions. You can configure default event tracking for supported components, or at the flow level, and also configure custom event tracking using the Custom Business Events Component.

Reviewer: ```suggestion
Mule applications can collect business event information about flows and message processors that handle your business transactions. You can configure default event tracking for supported components or entire flows, and you can also configure custom event tracking using the Custom Business Events component.
```

> For details, see xref:business-events.adoc[Business Events] documentation.

Reviewer: ```suggestion
For details, see the xref:business-events.adoc[Business Events] documentation.
```
